modules/farm_fd2/src/entrypoints/transplanting/lib.submit.unit.cy.js

```diff
@@ -0,0 +1,410 @@
+import { lib } from './lib.js';
+import * as farmosUtil from '@libs/farmosUtil/farmosUtil';
+
+describe('Submission using the transplanting lib.', () => {
+  let form = {
+    cropName: 'BROCCOLI',
+    picked: [
+      {
+        trays: 1,
+        data: {
+          log_id: '1',
+          log_uuid: 'ca7b2961-72da-4469-a2f1-af2abfc60082',
+          asset_id: '33',
+          asset_uuid: '0d0c5dd3-8888-4931-9f8c-e2998e7e9557',
+          date: '2019-03-11',
+          user: 'admin',
+          crop: 'BROCCOLI',
+          trays_location: 'CHUAU',
+          asset_locations: 'CHUAU',
+          total_trays: 4,
+          available_trays: 4,
+          tray_size: 128,
+          seeds_per_cell: 1,
+          total_seeds: 512,
+          notes: 'First broccoli tray seeding.',
+        },
+      },
+    ],
+    transplantingDate: '1950-01-02',
+    location: 'ALF',
+    beds: ['ALF-1', 'ALF-3'],
+    bedFeet: 100,
+    bedWidth: 60,
+    rowsPerBed: '1',
+    equipment: ['Tractor'],
+    depth: 6,
+    speed: 5,
+    comment: 'A comment',
+  };
+
+  let fieldMap = null;
+  let categoryMap = null;
+  let result = null;
+
+  before(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+
+    cy.wrap(farmosUtil.getFieldNameToAssetMap()).then((map) => {
+      fieldMap = map;
+    });
+
+    cy.wrap(farmosUtil.getLogCategoryToTermMap()).then((map) => {
+      categoryMap = map;
+    });
+
+    cy.wrap(lib.submitForm(form), { timeout: 10000 }).then((res) => {
+      result = res;
+    });
+  });
+
+  beforeEach(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+  });
+
+  afterEach(() => {
+    cy.saveLocalStorage();
+    cy.saveSessionStorage();
+  });
+
+  it('Check the parents', () => {
+    cy.wrap(farmosUtil.getPlantAsset(result.transplantingAsset.id)).then(
+      (plantAsset) => {
+        expect(plantAsset.relationships.parent[0].id).to.equal(
+          result.parents[0].id
+        );
+        expect(plantAsset.relationships.parent[0].type).to.equal(
+          'asset--plant'
+        );
+      }
+    );
+  });
+
+  it('Check the tray inventory quantity--standard assets', () => {
+    // cy.wrap(farmosUtil.getSeedlings('BROCCOLI')).then((seedlings) => {
+    //   console.log(seedlings);
+    //   expect(
+    //     seedlings[0].total_trays -
+    //       result.trayInventoryQuantities[0].attributes.value.decimal
+    //   ).to.equal(3);
+    // });
+    cy.wrap(
+      farmosUtil.getStandardQuantity(result.trayInventoryQuantities[0].id)
+    ).then((inventoryQuantity) => {
+      expect(inventoryQuantity.type).to.equal('quantity--standard');
+      expect(inventoryQuantity.attributes.measure).to.equal('count');
+      expect(inventoryQuantity.attributes.value.decimal).to.equal('1');
```

Comment:
Use `form` data when possible. For example, here use `form.picked[0].trays` instead of hard coding `1`.